sustainability_stock/models/stock_picking.py

```diff
@@ -0,0 +1,270 @@
+        )
+        unit_name = units[unit_select_id]
+
+        origin = picking.picking_type_id.warehouse_id.partner_id.with_context(
+            show_address=True, address_inline=True
+        ).display_name
+        destination = picking.partner_id.with_context(
+            show_address=True, address_inline=True
+        ).display_name
+
+        data = {
+            "route": [
+                {"location": {"query": origin}},
+                {"transport_mode": company.carbon_freight_transport_mode},
+                {"location": {"query": destination}},
+            ],
+            "cargo": {
+                "weight": picking.shipping_weight,
+                "weight_unit": unit_name,
+            },
+        }
+
+        error_message = _(
+            "An error occurred while retrieving carbon emissions data. If the issue persists, contact support."
+        )
+
+        try:
+            response = requests.post(url, json=data, headers=headers, timeout=30)
+
+            if response.status_code >= 400:
+                response_data = response.json()
+                logging.error(f"HTTP {response.status_code} {response.text}")
+                detailed_error = response_data.get("message", error_message)
+                return None, self._carbon_display_notification(
+                    detailed_error, sticky=True
+                )
+
+            return response.json().get("co2e", 0), None
+
+        except requests.exceptions.RequestException as err:
+            logging.error(f"Request error: {err}")
+            return None, self._carbon_display_notification(error_message)
+
+    @api.depends("state")
+    def _compute_carbon_debt(self) -> dict | None:
+        """
+        Computes and sets the carbon debt (CO2 emissions) for stock pickings in 'done' state.
+        Creates or updates accounting entries for the carbon emissions.
+        """
+        company = self.env.company
+        carbon_freight_max_delivery_count = company.carbon_freight_max_delivery_count
+
+        if len(self) > carbon_freight_max_delivery_count:
+            return self._carbon_display_notification(
+                _(
+                    f"You cannot process more than {carbon_freight_max_delivery_count} deliveries at once. Please edit the setting to increase the limit if needed."
+                )
+            )
+
+        carbon_freight_product_upstream = company.carbon_freight_product_upstream
+        carbon_freight_product_downstream = company.carbon_freight_product_downstream
+        carbon_freight_journal_id = company.carbon_freight_journal_id
+        carbon_freight_account_id = company.carbon_freight_account_id
+
+        for picking in self:
+            if picking.state != "done":
+                return self._carbon_display_notification(
+                    _(
+                        "Carbon emission can only be computed for pickings with state 'done'."
+                    )
+                )
```

Comment:
Maybe do this test before the max delivery count with a filtered, you could then output a notification with the pickings that are not computed and continue with the other ones ?

Reply:
showing the name of each picking would be too much: let's imagine we have a limit of 10 and you recompute 30 pickings, the message would be too big. instead i could display how many pickings were not computed